Fix new themes ready for packagist

RCheesley opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅ <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | 
| Deprecations?                          | 
| BC breaks? (use the c.x branch)        | 
| Automated tests included?              | <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

## Description

In https://github.com/mautic/mautic/pull/14155 we haven't been correctly set up each theme with a separate repo and Packagist package or configured the git split settings, so they are not going to be picked up by Composer installations. 

Each one should have a Packagist package, connected to the repository, and that should be pulled in [here](https://github.com/mautic/mautic/blob/5.x/.github/subtree-splitter-config.json) (which controls the git split, pushing changes / tags etc to the themes in core into the relevant repositories) and marked in the Composer file here: https://github.com/mautic/mautic/blob/5.x/composer.json#L53 this way they can be selectively added or removed from the build and in the future managed via the marketplace. 

This PR:

* Adds the git split configuration to connect the theme folder in mautic/mautic with the individual theme repos on GitHub
* Adds a GitHub Workflow for each sub-repo which we use to auto-close any PRs accidentally made in the sub-repos

Once this is merged, the sub-repos should be populated by the git split GitHub Action and then we can add the packages to composer.json (because we can't add an empty repo on Packagist - it requires the composer.json file).

### 📋 Steps to test this PR:

Please do code review as it won't be fully working until we get the Composer part sorted out.

The theme repos can be found using this query: https://github.com/orgs/mautic/repositories?type=all&q=theme.

RCheesley committed:
Add new themes to git split configurations

RCheesley committed:
add the GH workflow files to close pull requests on theme repos

RCheesley committed:
Add new themes to the workflow

matbcvo commented on .github/subtree-splitter-config.json:
```diff
@@ -169,6 +169,72 @@
+            "target": "git@github.com:mautic/theme-connect-through-content.git"
+        },
+        {
+            "name": "theme-education",
```
Rename from `theme-education` to `theme-educate`

https://github.com/mautic/mautic/blob/b0d683f77b9d968bba72c343db642e703ac6e8aa/themes/_educate/composer.json#L2

escopecz replied:
Yep, once this is done this should be ready to merge.

matbcvo commented on .github/subtree-splitter-config.json:
```diff
@@ -169,6 +169,72 @@
+        },
+        {
+            "name": "theme-education",
+            "directory": "themes/education",
```
Rename from `themes/education` to `themes/educate`

matbcvo commented on .github/subtree-splitter-config.json:
```diff
@@ -169,6 +169,72 @@
+        {
+            "name": "theme-education",
+            "directory": "themes/education",
+            "target": "git@github.com:mautic/theme-education.git"
```
Rename from `theme-education.git` to `theme-educate.git`
Also rename repository from `mautic/theme-education` to `mautic/theme-educate`

https://github.com/mautic/mautic/blob/b0d683f77b9d968bba72c343db642e703ac6e8aa/themes/_educate/composer.json#L2

RCheesley committed:
Fix incorrect reference to theme-education -> theme-educate

RCheesley commented:
Thanks for the review folks, grateful for another pair of eyes! 👀 

I've addressed that mistake and also changed the name of the [theme repository](https://github.com/mautic/theme-educate).

RCheesley committed:
Merge branch '5.x' into fix-new-themes-packagist

escopecz reviewed (approved):
Thanks! 👍
